### PR TITLE
fix claims check in auth middleware

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
The claims check in the auth middleware was switched around: in effect,
it checked that we did not receive _more_ than we asked for, rather than
checking we receive _at least_ what we asked for.

Of course this would still not let anyone run any trigger without the
proper access token, but it would let people list running triggers and
request (or stop) trigger executions.

CHANGELOG_BEGIN
- Fix a bug in the auth middleware where insufficient credentials could
  still give access to list of running triggers.
CHANGELOG_END

### CODE REVIEW

File: triggers/service/auth/src/test/scala/com/daml/auth/middleware/oauth2/TestMiddleware.scala [hunk 1]
```diff
@@ -96,12 +96,42 @@ class TestMiddleware extends AsyncWordSpec with TestFixture with SuiteResourceMa
...
+        aliceA_bobA <- r("Alice", "Bob")()
+        aliceA_bobR <- r("Alice")("Bob")
+        aliceR_bobA <- r("Bob")("Alice")
+        aliceR_bobR <- r()("Alice", "Bob")
+        bobA <- r("Bob")()
+        bobR <- r()("Bob")
+        bobAR <- r("Bob")("Bob")
       } yield {
-        assert(result == None)
+        assert(aliceA.isEmpty)
```
Comment (cocreature) [inline triggers/service/auth/src/test/scala/com/daml/auth/middleware/oauth2/TestMiddleware.scala:126]: `aliceA shouldBe None` gives you a better error if it fails iirc.

**Review by cocreature — APPROVED** [review]
uff, that’s bad. Thanks for catching & fixing it

File: triggers/service/auth/src/test/scala/com/daml/auth/middleware/oauth2/TestMiddleware.scala (hunk 1 above)
Comment (garyverhaegen-da) [reply triggers/service/auth/src/test/scala/com/daml/auth/middleware/oauth2/TestMiddleware.scala]: Done.

**Review by aherrmann-da — APPROVED** [review]
Thank you for catching this!

**Review by stefanobaghino-da — COMMENTED** [review]
👍🏻 🙇🏻

**Pull request merged**